diagrams/host_id_with_api_key_workflow.txt

```diff
@@ -0,0 +1,20 @@
+@startuml
+
+hide footbox
+
+participant "Puppet Master" as Master
+participant "Puppet Agent" as Agent
+participant Conjur
+
+Agent->Agent: Gather Facts
+Agent->Master: Facts
+Master->Conjur: Authenticate (with API Key)
+Conjur->Master: Auth Token
```

Comment:
my understanding was that in the course of gathering facts, the agent sends an authn request to conjur. it then sends the facts (including its conjur access token) to master

it _may_ be the case when the server has the conjur config, the server sends the conjur authn request. it would be good to better understand this.

Reply:
I do have the workflow that you're describing in another diagram. See `windows_pre_provision_workflow.txt`:
    https://github.com/cyberark/conjur-puppet/pull/138/files#diff-1d208015fa6d18f35ecc335493ea4a53R10.
In that diagram, the agent is pre-provisioned with the API key (either in a conjur identity file or in WinCred), so it does the authn request, gets the auth token, and passes that to the Puppet master.

In this diagram (host_id_with_api_key_workflow.txt), the intent is to show the workflow where the API key is stored in the Puppet manifest. My understanding (maybe @sgnn7 can confirm or deny) is that in this case the agent doesn't have immediate access to the API key or anything else in the Puppet manifest, until it sends facts to the server (and gets a catalog in return).

Reply:
I think I fixed this up. Marking as resolved to not keep the PR in stale condition.